diff --git a/Serializer.php b/Serializer.php
index 3f2461cf96..8e92abe29c 100644
--- a/Serializer.php
+++ b/Serializer.php
@@ -157,6 +157,10 @@ class Serializer implements SerializerInterface, ContextAwareNormalizerInterface
         }
 
         if (\is_array($data) || $data instanceof \Traversable) {
+            if ($data instanceof \Countable && 0 === $data->count()) {
+                return $data;
+            }
+
             $normalized = [];
             foreach ($data as $key => $val) {
                 $normalized[$key] = $this->normalize($val, $format, $context);
